lib/plugin.zsh

```diff
@@ -39,6 +34,11 @@ geometry_plugin_register() {
     return 1
   fi
```

Comment:
Won't this always return false for pinned plugins? Here we check against `+plugin` but only ever register `plugin` in `_GEOMETRY_PROMPT_PLUGINS`